internal/events/event.go

```diff
@@ -245,7 +248,7 @@ func (e *Events) V2AddMetricsEvent(ctx context.Context, clusterID *strfmt.UUID,
 	e.v2SaveEvent(ctx, clusterID, hostID, infraEnvID, name, models.EventCategoryMetrics, severity, msg, eventTime, requestID, props...)
 }
 
-func (e Events) queryEvents(ctx context.Context, selectedCategories []string, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID) ([]*common.Event, error) {
+func (e Events) queryEvents(ctx context.Context, selectedCategories []string, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID, limit, offset *int64) ([]*common.Event, *int64, error) {
```

Comment:
no reason for totalRecords to be a pointer
```suggestion
func (e Events) queryEvents(ctx context.Context, selectedCategories []string, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID, limit, offset *int64) ([]*common.Event, int64, error) {
```